cura/Utils/Decorators.py

```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2019 Ultimaker B.V.
+# Cura is released under the terms of the LGPLv3 or higher.
+
+import functools
+
+
+##  Decorator for functions that belong to a set of APIs. For now, this should only be used for officially supported
+#   APIs, meaning that those APIs should be versioned and maintained.
+#
+#   \param since The earliest version since when this API becomes supported. This means that since this version, this
+#                API function is supposed to behave the same. This parameter is not used. It's just a documentation.
+#
+def api(since = "Unknown"):

```

Comment:
I think we should force a correct API version to be used here (especially since we want to use this in the future to auto generate documentation!)

Reply:
From the requirements:
> The decorator should just be simple, without any arguments, even lacking 'from' and 'to' versions:

So I'd say that the parameter should be removed alltogether

Reply:
I added a semantic version check in 08b1e0e.